ldhooks/hooks.go

```diff
@@ -0,0 +1,93 @@
+package ldhooks
+
+import (
+	"context"
+
+	"github.com/launchdarkly/go-sdk-common/v3/ldreason"
+)
+
+// Implementation Note: The UnimplementedHook struct is provided to simplify hook implementation. It should always
+// contain an implementation of all series and handler interfaces. It should not contain the Hook interface directly
+// because the implementer should be required to implement GetMetadata.
+
+// A Hook is used to extend the functionality of the SDK.
+//
+// In order to avoid implementing unused methods, as well as easing maintenance of compatibility, implementors should
+// compose the `UnimplementedHook`.
+//
+//	type MyHook struct {
+//	  UnimplementedHook
+//	}
+type Hook interface {
+	GetMetadata() HookMetadata
```

Comment:
```suggestion
	Metadata() HookMetadata
```

(no Get prefix is idiomatic)